reliability/core-it/src/androidTest/kotlin/com/datadog/android/core/integration/FeatureSdkCoreContextTest.kt

```diff
@@ -0,0 +1,638 @@
+                getMockServerWrapper().getServerUrl()
+            )
+        }.onEach { testedFeatureSdkCore.registerFeature(it) }
+
+        // When
+        val retrievedFeatures = registeredFeatures.mapNotNull { testedFeatureSdkCore.getFeature(it.name)?.unwrap() }
+
+        // Then
+        assertThat(retrievedFeatures).containsAll(registeredFeatures)
+    }
+
+    @Test
+    fun mustReturnTheCrashReportFeature_when_getFeature_crashReportEnabled() {
+        // Given
+        val fakeConfigCrashReportsEnabled = fakeConfiguration.copy(crashReportsEnabled = true)
+
+        // When
+        // stop the current instance
+        Datadog.stopInstance()
+        val internalSdkCore = Datadog.initialize(
+            ApplicationProvider.getApplicationContext(),
+            fakeConfigCrashReportsEnabled,
+            fakeTrackingConsent
+        ) as InternalSdkCore
+
+        // When
+        val crashReportsFeature = internalSdkCore.getFeature(CRASH_REPORTS_TASK_NAME)?.unwrap<Feature>()
+
+        // Then
+        checkNotNull(crashReportsFeature)
+    }
+
+    @Test
+    fun mustReturnEmptyFeatures_when_getFeature_crashReportNotEnabled() {
+        // Given
+        val fakeConfigCrashReportsNotEnabled = fakeConfiguration.copy(crashReportsEnabled = false)
+
+        // When
+        // stop the current instance
+        Datadog.stopInstance()
+        val internalSdkCore = Datadog.initialize(
+            ApplicationProvider.getApplicationContext(),
+            fakeConfigCrashReportsNotEnabled,
+            fakeTrackingConsent
+        ) as InternalSdkCore
+
+        // When
+        val crashReportsFeature = internalSdkCore.getFeature(CRASH_REPORTS_TASK_NAME)?.unwrap<Feature>()
+
+        // Then
+        assertThat(crashReportsFeature).isNull()
+    }
+
+    @Test
+    fun isThreadSafe_when_registerMultipleFeatures_fromDifferentThreads() {
+        // Given
+        val registeredFeatures = forge.aList(size = forge.anInt(min = 1, max = 10)) {
+            StubStorageBackedFeature(
+                forge,
+                forge.anAlphabeticalString(),
+                getMockServerWrapper().getServerUrl()
+            )
+        }
+        val countDownLatch = CountDownLatch(registeredFeatures.size)
+        registeredFeatures.forEach { feature ->
+            val thread = Thread {
+                testedFeatureSdkCore.registerFeature(feature)
+                countDownLatch.countDown()
+            }
+            thread.start()
+        }
+        countDownLatch.await(SHORT_WAIT_MS, TimeUnit.MILLISECONDS)
+
+        // When
+        val retrievedFeatures = registeredFeatures.mapNotNull { testedFeatureSdkCore.getFeature(it.name)?.unwrap() }
+
+        // Then
+        assertThat(retrievedFeatures).containsAll(registeredFeatures)
+    }
+
+    @Test
+    fun isThreadSafe_when_registerSameFeature_fromDifferentThreads() {
+        // Given
+        val countDownLatch = CountDownLatch(10)
+        val threads = mutableListOf<Thread>()
```

Comment:
seems it is not used? or you want to call `.join` on each? in this case latch is not needed

Reply:
yes was using it before but then switched to countDownLatch